design/open/lakefs-kv-execution-plan.md

```diff
@@ -0,0 +1,72 @@
+# lakeFS on KV - Execution Plan
+
+Key/value data storage package (kv) that will supply key/value access to data to replace the current storage and locking done by Postgres.
+High level overview of the interface described in https://github.com/treeverse/lakeFS/blob/ed55edc2c24b24ed4e6fcccad1ae95844d38b9ad/design/open/metadata_kv/index.md. The package will support multiple back-end implementations.
+
+## Transform from using db to kv package
+
+High level interface described in the kv proposal, we should consider taking the DynamoDB interface to address the way we communicate `set-if` functionality while marshal the same value without fetching data first.
+
+Key/value format can be done using ProtoBuf / JSON.
```

Comment:
It will be very helpful if we can have a discussion on this to understand the tradeoffs between using ProtoBuf / JSON